codec/decode.go

```diff
@@ -2202,6 +2232,14 @@ func setZero(iv interface{}) {
 }
 
 func (d *Decoder) decode(iv interface{}) {
+	d.remainingDepth--
+	if d.remainingDepth < 0 {
+		panic("max depth exceeded")
+	}
+	defer func() {
```

Comment:
Embarassingly, I forgot to increment `remainingDepth` on the way out! This manifested in test failures when I lowered the limit to 100.

Reply:
yeah in my tests for my prepass, I had tests that failed and succeeded based on being at or just below the threshold, useful to catch bugs like this one.